modules/dxp/apps/analytics/analytics-reports-layout-display-page-impl/src/main/java/com/liferay/analytics/reports/layout/display/page/internal/info/item/LayoutDisplayPageObjectProviderAnalyticsReportsInfoItemImpl.java

```diff
@@ -0,0 +1,424 @@
+			_getThemeDisplayOptional();
+
+		return themeDisplayOptional.map(
+			themeDisplay -> {
+				Optional<Layout> layoutOptional = _getLayoutOptional(
+					layoutDisplayPageObjectProvider);
+
+				return layoutOptional.map(
+					layout -> {
+						HttpServletRequest httpServletRequest =
+							themeDisplay.getRequest();
+
+						LayoutDisplayPageObjectProvider<?>
+							initialLayoutDisplayPageObjectProvider =
+								(LayoutDisplayPageObjectProvider<?>)
+									httpServletRequest.getAttribute(
+										LayoutDisplayPageWebKeys.
+											LAYOUT_DISPLAY_PAGE_OBJECT_PROVIDER);
+
+						httpServletRequest.setAttribute(
+							LayoutDisplayPageWebKeys.
+								LAYOUT_DISPLAY_PAGE_OBJECT_PROVIDER,
+							layoutDisplayPageObjectProvider);
+
+						String completeURL = _portal.getCurrentCompleteURL(
+							httpServletRequest);
+
+						try {
+							String canonicalURL = _portal.getCanonicalURL(
+								completeURL, themeDisplay, layout, false,
+								false);
+
+							LayoutSEOLink layoutSEOLink =
+								_layoutSEOLinkManager.getCanonicalLayoutSEOLink(
+									layout, locale, canonicalURL,
+									_portal.getAlternateURLs(
+										canonicalURL, themeDisplay, layout));
+
+							return layoutSEOLink.getHref();
+						}
+						catch (PortalException portalException) {
+							_log.error(portalException, portalException);
+
+							return StringPool.BLANK;
+						}
+						finally {
+							httpServletRequest.setAttribute(
+								LayoutDisplayPageWebKeys.
+									LAYOUT_DISPLAY_PAGE_OBJECT_PROVIDER,
+								initialLayoutDisplayPageObjectProvider);
+						}
+					}
+				).orElse(
+					StringPool.BLANK
+				);
+			}
+		).orElse(
+			StringPool.BLANK
+		);
+	}
+
+	@Override
+	public Locale getDefaultLocale(
+		LayoutDisplayPageObjectProvider layoutDisplayPageObjectProvider) {
+
+		return Optional.ofNullable(
+			_groupLocalService.fetchGroup(
+				layoutDisplayPageObjectProvider.getGroupId())
+		).map(
+			Group::getDefaultLanguageId
+		).map(
+			LocaleUtil::fromLanguageId
+		).orElseGet(
+			LocaleUtil::getDefault
+		);
+	}
+
+	@Override
+	public Date getPublishDate(
+		LayoutDisplayPageObjectProvider layoutDisplayPageObjectProvider) {
+
+		try {
+			ClassName className = _classNameLocalService.getClassName(
+				layoutDisplayPageObjectProvider.getClassNameId());
+
+			InfoItemFieldValuesProvider infoItemFieldValuesProvider =
+				_infoItemServiceTracker.getFirstInfoItemService(
+					InfoItemFieldValuesProvider.class,
+					className.getClassName());
+
+			return Optional.ofNullable(
+				infoItemFieldValuesProvider.getInfoItemFieldValue(
+					layoutDisplayPageObjectProvider.getDisplayObject(), "publishDate")
+			).filter(
+				infoFieldValue -> {
+					InfoField infoField = infoFieldValue.getInfoField();
+
+					return Objects.equals(
+						infoField.getInfoFieldType(),
+						DateInfoFieldType.INSTANCE);
+				}
+			).map(
+				infoFieldValue -> (Date)infoFieldValue.getValue()
+			).orElseGet(
+				() -> new Date()
```

Comment:
Here, you can be replaced with method reference

Reply:
For now it is, for now, today will be the publish date, but we will fix it on the lima side :)